src/tpm2_wrap.c

```diff
@@ -323,6 +409,94 @@ int wolfTPM2_ReadPublicKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
     return rc;
 }
 
+/* primaryHandle must be owner or platform hierarchy */
+/* Owner    Persistent Handle Range: 0x81000000 to 0x817FFFFF */
+/* Platform Persistent Handle Range: 0x81800000 to 0x81FFFFFF */
+int wolfTPM2_NVStoreKey(WOLFTPM2_DEV* dev, TPM_HANDLE primaryHandle,
+    WOLFTPM2_KEY* key, TPM_HANDLE persistentHandle)
+{
+    int rc;
+    EvictControl_In in;
+
+    if (dev == NULL || key == NULL ||
+        (primaryHandle != TPM_RH_OWNER && primaryHandle != TPM_RH_PLATFORM) ||
+        persistentHandle < PERSISTENT_FIRST ||
+        persistentHandle > PERSISTENT_LAST) {
+        return BAD_FUNC_ARG;
+    }
+
+    /* if key is already persistent then just return success */
+    if (key->handle.hndl == persistentHandle)
+        return TPM_RC_SUCCESS;
+
+    /* clear auth */
+    XMEMSET(&dev->session[0].auth, 0, sizeof(dev->session[0].auth));
+
+    /* Move key into NV to persist */
+    XMEMSET(&in, 0, sizeof(in));
+    in.auth = primaryHandle;
+    in.objectHandle = key->handle.hndl;
+    in.persistentHandle = persistentHandle;
+
+    rc = TPM2_EvictControl(&in);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("TPM2_EvictControl failed %d: %s\n", rc, wolfTPM2_GetRCString(rc));
```

Comment:
It may be best with this and other printf calls to use something along the lines of XPRINTF instead?

Reply:
Good suggestion, but I will defer to future PR. For now requiring stdlib printf is sufficient.